clang/test/CIR/CallConvLowering/AArch64/aarch64-cc-structs.c

```diff
@@ -411,7 +410,7 @@ void qux(void) {
 // LLVM: void @qux
 // LLVM: %[[#V1:]] = alloca ptr, i64 1, align 8
 // LLVM: %[[#V2:]] = alloca i64, i64 1, align 8
-// LLVM: store ptr getelementptr (%struct.PackedS2, ptr @g, i64 1), ptr %[[#V1]], align 8
+// LLVM: store ptr getelementptr inbounds nuw (i8, ptr @g, i64 6), ptr %[[#V1]], align 8
```

Comment:
Great, thanks for you fixing this! Out of curiosity, did you double check that the LLVM output is the same if you try to codegen the same C sources directly to LLVM without CIR?

Reply:
Good question. The LLVM output without CIR seems to not include the constant folding behaviour. Unfortunately I didn't figure out why this diverging behaviour happens.

The LLVM output without CIR seems to sometimes include `inrange` arguments, which might have prevented the constant folding(?). I'm giving my best guess here.

Reply:
It's fine if it differs on the extra annotations (e.g. inrange which we haven't implemented), the important parts are the actual indicies and placement of nuw/nsw. Do these look sound to you?

Reply:
I think I'll need another qualified pairs of eyes to have a look --- I'm new to clang/LLVM and still trying to figure things out. The no-clangir path doesn't seem to consistently add `nuw`s, which might be somewhat concerning.

Reply:
I'll post compiler explorer links for some examples.

Reply:
> I think I'll need another qualified pairs of eyes to have a look

no problemo, I can help out.

> The no-clangir path doesn't seem to consistently add nuws

This is fine, it's not always needed, we might be over-emitting.